src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -208,6 +211,34 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         });
     }
 
+    Control::TermControl TermControl::AttachContent(Control::ControlInteractivity content, const Microsoft::Terminal::Control::IKeyBindings& keyBindings)
+    {
+        const auto term{ winrt::make_self<TermControl>(content) };
+        term->_AttachDxgiSwapChainToXaml(reinterpret_cast<HANDLE>(term->_core.SwapChainHandle()));
+        content.Reparent(keyBindings);
+
+        // Initialize the terminal only once the swapchainpanel is loaded - that
+        //      way, we'll be able to query the real pixel size it got on layout
+        auto r = term->SwapChainPanel().LayoutUpdated(winrt::auto_revoke, [term](auto /*s*/, auto /*e*/) {
+            // Replace the normal initialize routine with one that will allow up
+            // to complete initialization even though the Core was already
+            // initialized.
+            if (term->_InitializeTerminal(true))

```

Comment:
Yikes! Do we know whether ControlInteractivity/ControlCore are safe for multiple initialization?
Also, why is this called `AttachContent` when it seems to be more like, `NewControlByAttachingContent`? Should some of these things be in a private member function? I'm weirded out by you using `term->` so much to poke inside a newly constructed thing